pkg/security/tests/dns_resolver_test.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux && functionaltests
+
+// Package tests holds tests related files
+package tests
+
+// Package tests holds tests related files
+import (
+	"github.com/DataDog/datadog-agent/pkg/config/env"
+	sprobe "github.com/DataDog/datadog-agent/pkg/security/probe"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"net/netip"
+	"testing"
+	"time"
+)
+
+func TestDNSResolver(t *testing.T) {
+	SkipIfNotAvailable(t)
+	checkNetworkCompatibility(t)
+
+	if testEnvironment != DockerEnvironment && !env.IsContainerized() {
+		if out, err := loadModule("veth"); err != nil {
+			t.Fatalf("couldn't load 'veth' module: %s,%v", string(out), err)
+		}
+	}
+
+	test, err := newTestModule(t, nil, nil, withStaticOpts(testOpts{networkIngressEnabled: true}))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer test.Close()
+	p, _ := test.probe.PlatformProbe.(*sprobe.EBPFProbe)
+
+	// Makes a DNS query for a couple hostnames on a list and checks
+	// if the resolver saved all of them on the cache
+	t.Run("saves-hostname-for-all-ips", func(t *testing.T) {
+		// This test contains a 1 second backoff, and tries 10 times until it fails
+		attempts := 10
+
+		savedAllHostnamesFor := func(hostname string, ipAddresses []net.IP) {
+			assert.GreaterOrEqual(t, len(ipAddresses), 1)
+
+			for _, ipAddress := range ipAddresses {
+				for ; attempts != 0; attempts-- {
+					nip, _ := netip.AddrFromSlice(ipAddress)
+					list := p.Resolvers.DNSResolver.HostListFromIP(nip)
+
+					if len(list) != 0 {
+						assert.Equal(t, hostname, list[0])
```

Comment:
If I understand correctly, the list can also contain some cname right? should we do a slices.Contains() here instead of looking only at the first entry?